### ISSUE TITLE
Template dropdown validation does not work

### ISSUE THREAD

**Issue by FlorianCassayre** [camunda/camunda-modeler#3567]
### Describe the bug

The `constraints` field is not respected on properties of type `Dropdown`.

Here is a small template for reproducibility: 
```json
[
  {
    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema/resources/schema.json",
    "name": "Test",
    "id": "test",
    "version": 1,
    "appliesTo": [
      "bpmn:Task"
    ],
    "properties": [
      {
        "label": "Test",
        "type": "Dropdown",
        "choices": [
          { "name": "Label", "value": "value" }
        ],
        "constraints": {
          "notEmpty": true
        },
        "binding": {
          "type": "property",
          "name": "attribute"
        }
      }
    ]
  }
]
```

Then when I use this template, the Modeler does not validate that the dropdown must not be empty, despite the `"constraints": { "notEmpty": true }` property:

![image](https://user-images.githubusercontent.com/9027075/233940636-7cb5a11b-85ad-4da9-abac-c4353b293f5a.png)

For comparison, here is how properties of type `String` behave:

![image](https://user-images.githubusercontent.com/9027075/233942492-7ddd9ef0-0a90-494f-a827-2df3d2bd797a.png)

A workaround is to set the initial value of the Dropdown, but it's not ideal.

### Steps to reproduce

1. Create the above template file, save it in your `.camunda/element-templates/` directory
2. Create a diagram
3. Add a task on the diagram
4. Use the defined template on that task
5. Observe that the dropdown is not properly validated

### Expected behavior

Since
```json
"constraints": {
    "notEmpty": true
}
```
is allowed on properties of type `Dropdown`, I expect them to work the same way as `String` properties.

### Environment

- Camunda Platform 7
- Camunda Modeler Version: v5.10.0 (latest as of 2023.04.24)

### Additional context

The problem occurs on both Platforms (7 & 8).

**Comment by nikku** [comment]
I can reproduce your issue. Moving to backlog for the time being.

**Comment by smbea** [comment]
@marstamm is this fixed upstream via https://github.com/bpmn-io/bpmn-js-properties-panel/pull/922?

**Comment by marstamm** [comment]
Yes, this is fixed upstream in `bpmn-js-properties-panel@1.24.0`.
